File: src/other/list_node.kt
package other

class ListNode(var `val`: Int) {
    var next: ListNode? = null

    override fun toString(): String {
        val sb = StringBuilder();
        var node = this.next
        sb.append("$`val`")

        while (node != null) {
            sb.append("->${node.`val`}")
            node = node.next
        }

        return sb.toString()
    }
}

fun ListNode.printList() {
    var root: ListNode? = this

    while (root != null) {
        print("${root.`val`} -> ")
        root = root.next
    }

    println("")
}

File: lib/other/list_node.dart
class ListNode {
  int val;
  ListNode? next;

  ListNode([this.val = 0, this.next]);

  @override
  String toString() {
    var str = "";
    var node = this.next;

    str += "$val";

    while (node != null) {
      str += "->${node.val}";
      node = node.next;
    }

    return str;
  }
}

extension ListNodeExtension on ListNode {
  void printList() {
    ListNode? root = this;

    var str = "";

    while (root != null) {
      str += "${root.val} -> ";
      root = root.next;
    }

    print(str);
  }
}
